### PR TITLE
Minor update SA-Med2d 20M dataset

### PR DESCRIPTION
**Pull request by anwai98**
Added an additional logic to choose a fraction of samples per dataset for segmentation, expect the datasets with a small size already.

### CODE REVIEW

File: torch_em/segmentation.py [hunk 1]
```diff
@@ -250,8 +250,11 @@ def default_segmentation_dataset(
     is_seg_dataset=None,
     with_channels=False,
     with_label_channels=False,
+    verify_paths=True,
 ):
-    check_paths(raw_paths, label_paths)
+    if verify_paths:
+        check_paths(raw_paths, label_paths)
```
Comment (anwai98) [inline torch_em/segmentation.py:256]: @constantinpape Apparently this was the most expensive part (the one in `ImageCollectionDatasets` costs almost nothing). With this, the entire dataset can be loaded in ~30 seconds! ;)

Thanks for the suggestion!

Comment (constantinpape) [reply torch_em/segmentation.py:256]: We should probably expose this in `default_segmentation_loader` / `default_segmentation_dataset` too.

**Review by constantinpape — COMMENTED** [review]
I actually spotted one more change; we should expose the `verify_path` option to the upstream convenience functions.

**Comment by anwai98** [comment]
Thanks for spotting! The argument is now exposed to both `default_segmentation_dataset` and `default_segmentation_loader` now. GTG from my side!

**Comment by anwai98** [comment]
> Should we set `verify_paths=False` when `default_segmentation_dataset` is called in `get_sa_med_dataset`?

Yes, this definitely makes sense to add it to the dataset itself. Thanks for the mention!

**Pull request merged**